docs/reference/operation-reference.md

```diff
@@ -1,160 +1,343 @@
+
+### `mapValues`
+
+Transforms the value of each record without changing the key.
 
 #### Parameters
 
-| Parameter | Type   | Required | Description             |
-|-----------|--------|----------|-------------------------|
-| `if`      | Object | Yes      | Specifies the condition |
+| Parameter | Type   | Required | Description                          |
+|-----------|--------|----------|--------------------------------------|
+| `mapper`  | Object | Yes      | Specifies how to transform the value |
 
-The `if` can be defined using:
+The `mapper` can be defined using:
 
-- `expression`: A simple boolean expression
-- `code`: A Python code block returning a boolean
+- `expression`: A simple expression
+- `code`: A Python code block
 
-#### Example
+##### **See it in action**:
 
-```yaml
-- type: filter
-  if:
-    expression: value.get("age") >= 18
-```
+- [Tutorial: Branching](../tutorials/intermediate/branching.md)
 
-```yaml
-- type: filter
-  if:
-    code: |
-      if value.get("status") == "ACTIVE" and value.get("age") >= 18:
-        return True
-      return False
-```
+### `mapKey`
+
+Transforms the key of each record without modifying the value.
+
+#### Parameters
+
+| Parameter | Type   | Required | Description                        |
+|-----------|--------|----------|-----------------------------------|
+| `mapper`  | Object | Yes      | Specifies how to transform the key |
+
+The `mapper` can be defined using:
+- `expression`: A simple expression returning the new key
+- `code`: A Python code block returning the new key
+
+##### **See it in action**:
+
+- [Tutorial: Branching](../tutorials/intermediate/branching.md)
 
 ### `flatMap`
 
-Transforms each record into zero or more records.
+Transforms each record into zero or more records, useful for splitting batch messages into individual records.
 
 #### Parameters
 
 | Parameter | Type   | Required | Description                                                  |
 |-----------|--------|----------|--------------------------------------------------------------|
 | `mapper`  | Object | Yes      | Specifies how to transform each record into multiple records |
 
-The `mapper` can be defined using:
+The `mapper` must specify:
 
-- `expression`: A simple expression returning a list of tuples (key, value)
-- `code`: A Python code block returning a list of tuples (key, value)
+- `resultType`: Format `"[(keyType,valueType)]"` indicating list of tuples
+- `code`: Python code returning a list of tuples `[(key, value), ...]`
 
 #### Example
 
-```yaml
-- type: flatMap
-  mapper:
-    code: |
-      result = []
-      for item in value.get("items", []):
-        result.append((item.get("id"), item))
-      return result
-```
+This example splits order batches containing multiple items into individual item records:
 
-### `map`
+??? info "Producer definition (click to expand)"
 
-Transforms both the key and value of each record.
+    ```yaml
```

Comment:
Same here, better to link to one of the tutoruals, or to provide a really short example here (or both), for an example I think this is too much